### PR TITLE
Add Aggregate Report Pipeline

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Move dependency report into aggregate report pipeline.
Run dependency report using downloaded artifacts from latest successful scheduled run of the net - core pipeline.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Azure SDK for .NET
 
-[![Packages](https://img.shields.io/badge/packages-latest-blue.svg)](https://azure.github.io/azure-sdk/releases/latest/dotnet.html) [![Dependencies](https://img.shields.io/badge/dependencies-analyzed-blue.svg)](https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-net/dependencies/dependencies.html)
+[![Packages](https://img.shields.io/badge/packages-latest-blue.svg)](https://azure.github.io/azure-sdk/releases/latest/dotnet.html) [![Dependencies](https://img.shields.io/badge/dependencies-analyzed-blue.svg)](https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-net/dependencies/dependencies.html) [![DepGraph](https://img.shields.io/badge/dependencies-graph-blue.svg)](https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-net/dependencies/InterdependencyGraph.html)
```
Comment (heaths) [inline README.md]: Nit: IIRC, this shows up as alternative text, so maybe spell out "Dependency Graph".

File: eng/pipelines/aggregate-report.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+trigger: none
+pr: none
+
+jobs:
+  - job: Generate
+    variables:
+      - template: templates/variables/globals.yml
+    pool:
+      vmImage: 'windows-2019'
```
Comment (heaths) [inline eng/pipelines/aggregate-report.yml]: Not a big issue, but if there's nothing Windows-specific, ubuntu-18.04 is much faster.

File: eng/pipelines/aggregate-report.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        workingDirectory: $(System.ArtifactsDirectory)/BuildArtifacts
+        displayName: Remove Track 1 Packages
+
+      - pwsh: |
+          mkdir "$(Build.ArtifactStagingDirectory)/reports"
+          Copy-Item -Path "$(Build.SourcesDirectory)/eng/common/InterdependencyGraph.html" -Destination "$(Build.ArtifactStagingDirectory)/reports/InterdependencyGraph.html"
+          & "tools/AnalyzeDeps/AnalyzeDeps.ps1" -PackagesPath "$(System.ArtifactsDirectory)/BuildArtifacts/*.nupkg" -LockfilePath "$(Build.SourcesDirectory)/eng/Packages.Data.props" `
+          -OutPath "$(Build.ArtifactStagingDirectory)/reports/dependencies.html" -DumpPath "$(Build.ArtifactStagingDirectory)/reports/data.js"
+        displayName: "Analyze dependencies"
+        condition: succeeded()
```
Comment (heaths) [inline eng/pipelines/aggregate-report.yml]: `succeeded()` is the default condition, so these instances here and below aren't necessarily needed.

**Review by heaths — APPROVED** [review]

File: eng/pipelines/aggregate-report.yml (hunk 3 above)
Comment (chidozieononiwu) [reply eng/pipelines/aggregate-report.yml]: Oh, I see thanks

File: eng/pipelines/aggregate-report.yml (hunk 2 above)
Comment (chidozieononiwu) [reply eng/pipelines/aggregate-report.yml]: Ok, AzureFileCopy is windows specific.

File: eng/pipelines/aggregate-report.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      vmImage: 'windows-2019'
+    steps:
+      - pwsh: mkdir $(System.ArtifactsDirectory)/BuildArtifacts
+        displayName: Create Directory
+
+      - task: DownloadPipelineArtifact@2
+        displayName: 'Download Pipeline Artifact'
+        inputs:
+          buildType: specific
+          project: $(TargetPipelineProjectId)
```
Comment (weshaggard) [inline eng/pipelines/aggregate-report.yml]: Which pipeline is this pulling? Do you know if it only pulls latest successful build?

File: eng/pipelines/aggregate-report.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+trigger: none
```
Comment (weshaggard) [inline eng/pipelines/aggregate-report.yml]: Both java and python have file and pipeline called `aggregate-reports` can you please update to match?

File: eng/pipelines/aggregate-report.yml (hunk 4 above)
Comment (chidozieononiwu) [reply eng/pipelines/aggregate-report.yml]: It pulls the latest successful master branch build of the net-core and downloads the artifacts called 'packages'.
I didn't use `allowPartiallySucceededBuilds` or `allowFailedBuilds` so there is not case where it should download an unsuccessful run.

File: eng/pipelines/aggregate-report.yml (hunk 5 above)
Comment (chidozieononiwu) [reply eng/pipelines/aggregate-report.yml]: I will update the name.

**Pull request merged**